### ISSUE TITLE
.dependabot/config.yml and forks

### ISSUE THREAD

**Issue by LeoColomb** [dependabot/dependabot-core#2198]
[`.dependabot/config.yml`](https://dependabot.com/blog/introducing-config-files/) is a great feature, thanks!

That said, I faced an unwanted behavior.
Considering two accounts with Dependabot enabled, if an account forks a repo of the other one containing a `.dependabot/config.yml` file and with outdated dependencies, Dependabot will be auto-activated and will open pull requests on the forked repo.

As a real example, I forked `dependabot-core` and got four PR in the minute.

**Comment by hazendaz** [comment]
When will the bug be fixed with forks?  It seems completely backwards that I have to delete a fork and recreate it.  I don't care to do that and this is an incredible mess.  I'm getting daily bast of PRs that is incredibly high for repos I work on and its too easy to accidently merge on the fork where I don't want the merge.  This has been an outstanding issue before V2 came into the picutre now for going on 2 years.  So please make this a priority.

**Comment by onedrawingperday** [comment]
@gohugoio recently added dependabot to its [master repo](https://github.com/gohugoio/hugo). 
Currently there are some 5.3 K forks of the main repo. Lots of these were created before v2 of .dependabot. 

You cannot really expect people to update to v2 by deleting their forks and associated pull requests history so that they stop receiving these unwanted Pull Requests.

Please do consider making this issue a priority.

**Comment by s-weigand** [comment]
IMHO the easyest fix for this issue would be to add a new field to `dependabot.yml` which might i.e. be called `repo`, which could lead to 3 different cases.
- *no repo field* => current behaviour
- *repo field value matches repo name* => do PR's
- *repo field value does NOT match repo name* => don't do PR's

That way you could get the following benefits:
- Backward compattiblety
- No unwanted PR's, since repo value and name mismatch, if it is a contrib fork (as I guess most are)
- Option to activate dependabot on forks (i.e. hard forks can just change the repo field value)

IMHO this is a case where users don't want all the automagic assumptions, but have full control.
When my dependabot Tesla is about to crash in tree, I want the option to deactivate the autopilot. 😉

**Comment by helfper** [comment]
Can we just get a `@dependabot ignore this repo` to make Dependabot go away from the repo/fork?

**Comment by cdrini** [comment]
Would love it if this was bumped in priority :) We got/merged a PR from dependabot https://github.com/internetarchive/openlibrary/pull/3831 4 days ago, and now presumably all up-to-date forks are getting update notifications :/ Mine certainly is!

**Comment by raisedadead** [comment]
We just enabled dependabot on our repository with ~25000 forks and we have ~900 dependencies. We did not know about this known issue. 

A heads up if this would mean any spikes on dependabot's backend and resulting support issues (which I imagine to increase over time as forks get updated).

I for one love what dependabot does and a config to ignore forks would be helpful to us maintainers and save wasted resources on dependabot's end aswell?

If there anything we can do to help with this, please let us know.

**Comment by mhutchinson** [comment]
Hi Dependabot team! Unfortunately, we've also fallen into this trap after upgrading our repos to use the configuration file. It sounds like a proper fix is some months away. In the interim, can I suggest:
 1. Referencing this Known Issue from the [config file documentation](https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates#about-the-dependabotyml-file)? This is a surprising thing to happen, and should be an active consideration for any repository maintainers that are upgrading to the new version
 2. Seeing whether there is a short-term fix that could help with manually disabling dependabot for a fork. I'd expect to be able to do this from the fork's Settings page. It's very surprising to me that I have dependabot running against my fork of a repo, yet when I look in the Settings, there are no integrations listed and all of the dependency options under `Configure security and analysis features` are listed as not enabled.

**Comment by patcon** [comment]
I tend to try to be positive in GitHub issues, because imho it's important we're all kind to one another as a first principle in this sort of work ❤️  Having said that, this outstanding issue feels _stunningly_ negligent on behalf of the Dependabot and GitHub team. Many people cannot simply recreate the fork. Pls consider putting some time into this bug

**Comment by daniel-beck** [comment]
As a workaround I create a new branch in my forks without `.dependabot.yml` (`git checkout --orphan` is cleanest IMO, but probably optional), and made that default branch of my fork(s). Dependabot only cares about the default branch contents even if [the docs](https://docs.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates#about-the-dependabotyml-file) don't explicitly say so.

Obviously doesn't scale to other people who fork your project's main repo, and could be confusing for people accessing the fork, but this could be useful for folks with forks that are only used by them to open pull requests from.

**Comment by blueyed** [comment]
The docs state that it would have to be enabled explicitly with forks based on dependabot v2, but that's not the case for me at least (where the fork was created at the beginning of the year): https://docs.github.com/en/code-security/supply-chain-security/enabling-and-disabling-version-updates#enabling-version-updates-on-forks

It would be helpful already if there was a way to disable it from there also, and/or the `@dependabot ignore this repo` command that was suggested.

**Comment by famod** [comment]
I honestly cannot imagine how many cpu (and brain) cycles have been wasted by this.
I keep on commenting `@dependabot ignore this dependency` in all PRs the bot is creating in my fork of Quarkus and maybe one day I will have ignored them all. 😒 

Any yes, I have recreated my fork but as soon as I push to `main` (which can make sense, at least for Quarkus it does), the bot is back.
I certainly won't pollute my fork's main by deleting the file...

PS: Sorry for my "mood". The bot is free of charge after all, but is adding at least `@dependabot ignore this repo` so difficult?

**Comment by trivikr** [comment]
Bumping this, as it has been 3+ months since the comment about the fix in related issue https://github.com/dependabot/dependabot-core/issues/2804#issuecomment-769954600

![Screen Shot 2021-05-04 at 7 53 25 AM](https://user-images.githubusercontent.com/16024985/117023328-dd79d180-acad-11eb-8c86-bebeb8ef918b.png)

**Comment by blueyed** [comment]
Is this a duplicate of #2804 then after all (at least)?

**Comment by rgommers** [comment]
> We also want to allow running on forks so won't be a super quick fix on our side. 

It's 2.5 years later now ..... how many more incoming links or thumbs-up emoji's do you need @feelepxyz?

**Comment by nstringham** [comment]
I like the `@dependabot ignore this repo` idea it would also be nice if dependabot could be disabled from `https://github.com/<user>/<repo>/network/updates`

**Comment by javierm** [comment]
Our workaround so far to avoid spamming our 800+ forks:

* In the `master` branch, we've got no dependabot config file
* In another branch, we've got a dependabot config file with the `target-branch: master` option

When we want to run dependabot, we make the latter branch the default one, and then make the `master` branch the default branch again.

Of course, now it's August and most people potentially interested in forking our project are on holidays, so the risk of someone forking the project while we're changing the default branch is very low :thinking:. Might not work for projects people are forking constantly.

**Comment by GoliathLabs** [comment]
@feelepxyz any updates on this? I really like dependabot and recommend it to the projects I use in my own infrastructure to make updating dependencies easier. However, most of them turn it off after a while because people with forks complain about dependabot's PR spam. I think that after 3 years now, this problem should be worked on, as some projects would very well want to use dependabot but don't because of this situation. @nstringham has pointed out a good way to compensate for this. If this is resolved, dependabot would get many more users and the debates and conflicts regarding the use of dependabot for a project would be much easier to resolve.

**Comment by hugovk** [comment]
I forgot one project uses dependabot and made the mistake of updating my fork from `upstream/main` and got 26 spam PRs:

<img width="1222" alt="image" src="https://user-images.githubusercontent.com/1324225/156926470-49835e57-1aeb-4222-b995-cd5185b3f70f.png">

**Comment by mikehardy** [comment]
https://docs.renovatebot.com/configuration-options/#includeforks

Renovate by default skips all forks - looks like a good alternative if change in the repo you manage is possible. I've seen renovate used to good effect in the ACRA repo and will consider it for my repos since the fork PR thing is kind of a showstopper as it induces a lot of contributor friction in fork-based repos.

**Comment by patcon** [comment]
Renovate is also fully open source

**Comment by jeffwidman** [comment]
👋 Hey there! This is super annoying, and very much on our radar. Unfortunately, I can't give any promises on timeline etc, but it is something we plan to fix.

However, it's also tracked over in https://github.com/dependabot/dependabot-core/issues/2804, so I'm going to close this as a duplicate. I realize this one is earlier, but the other one is the one we're tracking internally on some project boards, so just easier to keep that one as the canonical reference ticket.

**Comment by koppor** [comment]
> 👋 Hey there! This is super annoying, and very much on our radar. Unfortunately, I can't give any promises on timeline etc, but it is something we plan to fix. However, it's also tracked over in #2804, so I'm going to close this as a duplicate. I realize this one is earlier, but the other one is the one we're tracking internally on some project boards, so just easier to keep that one as the canonical reference ticket.

@jeffwidman The mentioned issue is closed with > 1.700 comments. Could you summarize the result for us?

**Comment by hugovk** [comment]
Summary:

* For new forks (after 7th November 2022), it is already disabled.

* For old forks, you can delete and re-fork, or disable it via the settings at `/settings/security_analysis`:

![image](https://github.com/dependabot/dependabot-core/assets/1324225/1242821c-dff0-4b6c-966a-c7fede342309)
